docs/source/python/json.rst

```diff
@@ -0,0 +1,114 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. currentmodule:: pyarrow.json
+.. _json:
+
+Reading JSON files
+==================
+
+Arrow supports reading columnar data from JSON files.  A JSON file consists
+of multiple JSON objects, one per line, representing individual data rows.
```

Comment:
Does arrow supports only such line-delimited json files? (http://ndjson.org/, http://jsonlines.org/) 
Although it is probably rather common in certain file types (eg logs), it is not the standard "JSON file" (AFAIK). If so, that might be worth clarifying.

Reply:
Is there a standard "JSON file"? In any case, the clarification is just above, so I'm not sure what needs explaining.

Reply:
I'll try to make the wording a bit more explicit about this.

Reply:
Not too familiar with standards, but at least in pandas the default for a json file is to consist of a single json object (which can then still be in many different forms). That's also what the JSON Table Schema does.

But thanks for the update! My only remaining suggestion would be to explicitly use the term "newline-delimited JSON files" somewhere, as that seems to be the accepted term for this.

Reply:
Data warehousing systems generally only support newline delimited JSON, as it's the standard in "big data systems" e.g. for web application logging

https://cloud.google.com/bigquery/docs/loading-data-cloud-storage-json

but we'll have to support more things in Arrow